File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Apparel.java
package com.logistics.Components.Product;

public class Apparel extends Product {

    protected String material;

    public Apparel() {
        super();
        material = "";
    }

    public Apparel(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String material) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit);
        this.material = material;
    }

    public Apparel(Apparel other) {
        super(other);
        this.material = other.material;
    }

    @Override
    public Apparel clone() {
        return new Apparel(this);
    }

    //Getters
    public String getMaterial() { return material; }
    //Setters
    public void setMaterial(String material) { this.material = material; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nMaterial: " + material;
    }
}

File: Task 3/C++Stock Viewer/Product/Apparel.cpp
#include "Apparel.h"

Apparel::Apparel() : Product(){
    material = "";
};

Apparel::Apparel(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string material) 
        : Product(name, cost, supplier, discount_amount, packing_units, cost_per_unit){
    this->material = material; 
};

Apparel::Apparel(const Apparel& other) : Product(other) {
    this->material = other.material;
}

std::string Apparel::display_product_info(){
    return Product::display_product_info() +
            "\nMaterial: "+material;
}
